backend/workflow_manager/workflow/workflow_helper.py

```diff
@@ -549,24 +550,40 @@ def complete_execution(
     ) -> ExecutionResponse:
         if pipeline_id:
             logger.info(f"Executing pipeline: {pipeline_id}")
-            if not execution_id:
-                workflow_execution = (
-                    WorkflowExecutionServiceHelper.create_workflow_execution(
-                        workflow_id=workflow.id,
-                        single_step=False,
-                        pipeline_id=pipeline_id,
-                        mode=WorkflowExecution.Mode.QUEUE,
-                        execution_id=execution_id,
-                    )
+            # Create a new WorkflowExecution entity for each pipeline execution.
+            # This ensures every pipeline run is tracked as a distinct execution.
+            workflow_execution = (
+                WorkflowExecutionServiceHelper.create_workflow_execution(
+                    workflow_id=workflow.id,
+                    single_step=False,
+                    pipeline_id=pipeline_id,
+                    mode=WorkflowExecution.Mode.QUEUE,
+                    execution_id=str(uuid.uuid4()),
                 )
-                execution_id = workflow_execution.id
-            response: ExecutionResponse = WorkflowHelper.execute_workflow_async(
+            )
+            execution_id = workflow_execution.id
+            log_events_id = StateStore.get(Common.LOG_EVENTS_ID)
+            org_schema = connection.tenant.schema_name
+            execution_result = WorkflowHelper.execute_bin(
```

Comment:
@muhammad-ali-e does this become a synchronous call then? What happens in case of gateway timeouts?

Reply:
It already in asyncrounous behaviour as it is  a pipline

Reply:
Already in an asynchronous context